src/runtime/transformers/shiki/highlighter.ts

```diff
@@ -145,17 +145,24 @@ export const useShikiHighlighter = createSingleton((opts?: Exclude<ModuleOptions
   const getHighlightedAST = async (code: string, lang: Lang, theme: Theme, opts?: Partial<HighlighterOptions>): Promise<Array<MarkdownNode>> => {
     const lines = await getHighlightedTokens(code, lang, theme)
     const { highlights = [], colorMap = {} } = opts || {}
+    const lineBreak: MarkdownNode = { type: 'element', tag: 'br' }
 
-    return lines.map((line, lineIndex) => ({
+    const highlightedLines: MarkdownNode[] = lines.map((line, lineIndex) => ({
       type: 'element',
-      tag: 'div',
+      tag: 'span',
       props: {
         class: ['line', highlights.includes(lineIndex + 1) ? 'highlight' : ''].join(' ').trim(),
         line: lineIndex + 1
       },
       children: line.map(tokenSpan)
     }))
 
+    for (let i = highlightedLines.length - 1; i > 0; i--) {
+      highlightedLines.splice(i, 0, lineBreak)
+    }
```

Comment:
I believe we don't need to add `<br>` tag after each line. These lines will wrap inside `<pre>` tag which respects new lines `\n`

Simply changing `div` to `span` should do the trick

Reply:
But then we have the problem, that copying code by selection in Firefox removes the line breaks, again.
That was the reason to use divs instead of spans.

Reply:
Adding `<br>` tag will result in duplicate new lines.

<img width="972" alt="Screenshot 2023-04-17 at 14 23 31" src="https://user-images.githubusercontent.com/2047945/232482866-f778dd88-b6cd-45d7-80e6-e58682bad372.png">

I remember that, but I think we should fix that issue on `CopyButton`. WDYT?

Reply:
The copy button does not have that problem, because it gets the raw code.

Only copying by selection in Firefox is problematic, because spans styled as block don't get copied with newlines between the lines

Reply:
What about appending a line break char (`\n`) to the last token of each line, if there isn't any? Firefox should detect that

Reply:
I'm at it. Thanks

Reply:
Now using line break characters